preprocessors/togeojson-gpx.preprocessor.js

```diff
@@ -102,6 +102,14 @@ module.exports = function(infile, outdirectory, callback) {
       });
     });
 
+    var archivedOriginal = path.join(outdirectory, '/archived.gpx');
+    digest(infile, function(err, callback) {
+      fs.writeFile(archivedOriginal, fs.readFile(infile), function(err) {
```

Comment:
Be aware that `fs.readFile()` is asynchronous, therefore it expects a callback ([see docs](https://nodejs.org/api/fs.html#fs_fs_readfile_file_options_callback)).

If you wanted to read the file synchronous, you would need to run `fs.readFileSync()`. One way to debug this is to create a separate variable to hold the contents of `fs.readFileSync(infile)`, just before passing them into `fs.writeFile()`. This way you can easily console log and ensure you have the proper value.